### PR TITLE
Game slice widget type

### PR DESCRIPTION
**Pull request by pconerly**
Hi again! @Kashomon 

So, I'm gonna be blogging my game reviews, and I want a way to say: "write the move numbers from move X to move Y."  I'm trying to make diagrams like in here: http://gobase.org/studying/articles/matthews/fuseki/08/

I know that we have an existing way to do this by adding the labels, but it becomes a not-fun manual process.

So here's my change.  Here's what `GameSliceTester.html` looks like:

screen shot 2015-11-11 at 9 18 22 pm

### CODE REVIEW

File: src/compiled/glift_combined.js [hunk 1]
```diff
@@ -10565,6 +10578,59 @@ glift.controllers.BoardEditorMethods = {
...
+glift.controllers.GameSliceMethods = {
+
+  initialize: function(treepath) {
+    var rules = glift.rules;
+    var initTreepath = treepath || this.initialPosition;
+    this.treepath = rules.treepath.parsePath(initTreepath);
+
+    var initialPosition = this.treepath.length; // used later
+    var nextTreepath = this.drawTo - this.treepath.length;
+    if (this.nextMovesPath.length > 0) {
```
Comment (pconerly) [inline src/compiled/glift_combined.js]: Just some self review: I shouldn't overload `nextMovesPath`, I'd like to allow `drawTo` to contain a position or a variation.

**Comment by pconerly** [comment]
I'm currently using this and there are some bugs with trying to use the `nextMovesPath` anyway. I do want a way to pick a variation to go down, but it's broken currently...

**Comment by Kashomon** [comment]
Couple high level comments: Thanks, I think this does probably make sense to make its own widget, at least for now.  There's a good chance that something like GPub would be the long term solution, but it's not worth considering at the moment.

- I'm not sure I GameSlice is the best naming, but I'm having a hard time coming up with better naming.
- Could you write some tests?

**Comment by pconerly** [comment]
As far as names go, how about Sequence or GameFigure?  They're not much better but they might convey the idea.

What is GPub?  Is that the markdown/sgf book format you mentioned in https://github.com/Kashomon/glift/issues/97 ?

**Comment by Kashomon** [comment]
Ok, I'm going to merge this change in. Know that I see this game type as experimental. It's possible that I'll decide to combine this with the Example widget or rework this.

**Pull request merged**